website/guide.yml

```diff
@@ -5,24 +5,106 @@
+
+      - subcategory: "Inference Engine"
+        content: |
+          Inference Engines are optimized frameworks or platforms designed to execute machine learning models, particularly large language models (LLMs), with an emphasis on high performance, low latency, and scalability. These engines allow users to run inferences across various hardware platforms and deploy models efficiently. They handle tasks such as model loading, serving, optimization, and high-throughput execution, making them essential in AI and machine learning applications.
+
+      - subcategory: "Inference Platform"
+        content: |
+          Inference Platforms are specialized environments or infrastructures designed for deploying and managing large-scale AI inference tasks, particularly for Generative AI (GenAI) models and large language models (LLMs). These platforms focus on providing efficient resource management, scaling capabilities, and ease of use when serving models in production environments, often integrating Kubernetes for orchestration, auto-provisioning, and container-based deployments.
+
+      - subcategory: "LLM Router"
+        content: |
+          LLM routers are specialized systems designed to intelligently route, proxy, or switch between multiple large language model (LLM) providers based on factors like cost, latency, quality, or model capabilities. They help developers and organizations optimize API usage across different backends while maintaining a unified, often OpenAI-compatible, interface. Some also integrate guardrails and evaluation mechanisms to ensure safe and efficient AI interactions.
+
+      - subcategory: "Output"
+        content: |
+          Output refers to tools or frameworks that help generate structured or formatted results from large language models (LLMs). These tools are designed to provide well-organized and easily interpretable outputs, especially for complex text generation tasks.
+
+
+  - category: "Orchestration"
+    content: |
+      Orchestration in AI involves coordinating and managing the various components, workflows, agents, and services that make up complex AI systems.
+      It ensures that models, tools, APIs, agents, and data sources work seamlessly together, supporting scalability, automation, modularity, and efficient deployment from development to production environments.
+
+    subcategories:
+      - subcategory: "Agent"
+        content: |
+          An Agent is an autonomous or semi-autonomous system powered by large language models, integrating RAG, Function Calling, and MCP to understand user intent, retrieve external knowledge, invoke tools or APIs, and dynamically execute complex tasks with precise responses.
+
```

Comment:
`Agent` is now at the same level as the `Framework` and `Tools`, but in README.md, the `Framework` and `Tools` are a subcategory of `Agent`.
![CleanShot 2025-04-27 at 15 06 27@2x](https://github.com/user-attachments/assets/51bc0982-f524-4e55-91f4-679adda4959a)

Could you give me an example of what kinds of projects will be categorized as `Agent` but not as `Framework` or `Tools`?

Reply:
Agent now is the original agent.framework, and framework is the original application orchestration framework. I can't think of better names for them, the boundaries of  their concepts are really vague. Do you have other ideas?

Reply:
Rename framework to workflows